src/types/label.js

```diff
@@ -24,6 +24,13 @@ export default class LabelAnnotation extends Element {
     translate(ctx, this.getCenterPoint(), options.rotation);
     drawCallout(ctx, this);
     drawBox(ctx, this, options);
+    if (this.box) {
```

Comment:
I needed to store the box annotation dimensions in order to apply to clip. I don't like it too much.

Reply:
we cannot apply the clip to all annotations :(
Exceptions:
1. callout for LABEL annotation
2. arrowHeads for LINE annotation

Both above mentioned elements are NOT included in box model calculation.

Reply:
could pass chartArea to the parents?

Reply:
and then remove the outer clip, so annotations can draw outside chart area if needed...

Reply:
I have added a method to box annotation which returns the bounding box.